File: demo-wiremock/src/test/java/io/fries/wiremock/demo/wiremock/cucumber/wiremock/replay/ReplayMockServer.java
package io.fries.wiremock.demo.wiremock.cucumber.wiremock.replay;

import com.github.tomakehurst.wiremock.WireMockServer;
import io.fries.wiremock.demo.wiremock.cucumber.wiremock.MockServer;

public class ReplayMockServer implements MockServer {

    private final WireMockServer server;

    public ReplayMockServer(final WireMockServer server) {
        this.server = server;
    }

    @Override
    public void start() {
        server.start();
    }

    @Override
    public void stop() {
        server.stop();
    }
}

File: examples/demo-cucumber/src/test/kotlin/io/fries/proxito/demo/cucumber/glue/wiremock/replay/ReplayMockServer.kt
package io.fries.proxito.demo.cucumber.glue.wiremock.replay

import com.github.tomakehurst.wiremock.WireMockServer
import io.fries.proxito.demo.cucumber.glue.wiremock.MockServer
import io.fries.proxito.demo.cucumber.glue.world.World

class ReplayMockServer(
    private val world: World,
    private val server: WireMockServer
) : MockServer {

    override fun start() = with(server) {
        start()
        addMockServiceRequestListener { request, response ->
            world.tracker().track(request, response)
        }
    }

    override fun stop() = server.stop()
}